src/processor.ts

```diff
@@ -316,8 +321,12 @@ processor.run(new TypeormDatabase(), async (ctx) => {
                         break
                     // price oracle
                     case 'DiaOracleModule.UpdatedPrices':
-                        // Only processing these events if the current block is the 'head' of the chain
-                        if (ctx.isHead) {
+                        // Only processing these events once every CATCHUP_PRICE_UPDATE_PERIOD blocks or if the current block is the 'head' of the chain
+                        // CATCHUP_PRICE_UPDATE_PERIOD is used so that we don't have to process these events for every block but still maintain a fairly accurate price history
+                        if (
+                            block.height % catchupPriceUpdatePeriod === 0 ||
+                            ctx.isHead
```

Comment:
Nit: Maybe switch order because checking `ctx.isHead` is faster than doing the math operation.